src/sage/monoids/string_ops.py

```diff
@@ -94,24 +98,21 @@ def coincidence_discriminant(S,n=2):
         sage: coincidence_discriminant([ S[i:i+2] for i in range(len(S)-1) ])
         0.0827001855677322
     """
-    if not isinstance(S,(list,tuple)):
+    if not isinstance(S, (list, tuple)):
         raise TypeError("Argument S (= %s) must be a list or tuple" % S)
     if n != 2:
         raise ValueError("Argument n (= %s) is only implemented for n = 2" % n)
-    truth = True
-    for bool in ( isinstance(c,(str,StringMonoidElement)) for c in S ):
-        truth = truth and bool
+    truth = all(isinstance(c, (str, StringMonoidElement)) for c in S)
     if not truth:
```

Comment:
```suggestion
    if not all(isinstance(c, (str, StringMonoidElement)) for c in S):
```

Reply:
both variants have the same short-circuiting semantics, and I'd say there's no difference in readability